rewrite-java/src/main/java/org/openrewrite/java/RemoveAnnotationVisitor.java

```diff
@@ -36,7 +37,7 @@
 
         List<J.Annotation> leadingAnnotations = classDecl.getLeadingAnnotations();
         if (annotationRemoved != null) {
-            if (leadingAnnotations.get(0) == annotationRemoved || leadingAnnotations.size() == 1) {
+            if (leadingAnnotations.get(0) == annotationRemoved && leadingAnnotations.size() == 1) {

```

Comment:
I'm unsure if the OR in `if (leadingAnnotations.get(0) == annotationRemoved || leadingAnnotations.size() == 1)` was intentional.

The `leadingAnnotations.get(0) == annotationRemoved` only works if the size == 0. So, the condition may either be `leadingAnnotations.get(0) == annotationRemoved && leadingAnnotations.size() == 1` or `leadingAnnotations.size() == 1`.

I'm curious if the condition needs to be `leadingAnnotations.size() == 1` to work as a catch all incase the `leadingAnnotations.get(0) == annotationRemoved` fails?